shell/common/node_bindings.cc

```diff
@@ -237,14 +237,17 @@ GetAllowedDebugOptions() {
   if (electron::fuses::IsNodeCliInspectEnabled()) {
     // Only allow DebugOptions in non-ELECTRON_RUN_AS_NODE mode
     return {
-        "--inspect",          "--inspect-brk",
-        "--inspect-port",     "--debug",
-        "--debug-brk",        "--debug-port",
-        "--inspect-brk-node", "--inspect-publish-uid",
+        "--inspect",           "--inspect-brk",
+        "--inspect-port",      "--debug",
+        "--debug-brk",         "--debug-port",
+        "--inspect-brk-node",  "--inspect-publish-uid",
+        "--frozen-intrinsics",
     };
   }
   // If node CLI inspect support is disabled, allow no debug options.
-  return {};
+  return {
+      "--frozen-intrinsics",
+  };
```

Comment:
I'm not sure it fits into the same fuse, but I don't think we should allow this flag to be unpreventable.  It changes the behavior of the JS in the main process and I can imagine a world where that can be abused to result in different and malicious behavior.

I'm also not fully sure that this flag makes sense for us to support as a flag rather than a separate fuse style toggle.  It's very hard if not impossible to pre-specify a flag like this in a packaged application.

I think I'd rather see this implemented as a fuse that forces intrinsics to be frozen at package time rather than enabling it as a choice at runtime

Reply:
the intentional use is to pass it to the renderer process using `additionalArguments: ['--frozen-intrinsics']`

Reply:
IMO we either go the fuse route or document and limit it to just the renderer process then 🤔 .

I'd also note that `additionalArguments` is a flawed property nowadays (args aren't per-webcontents they are per-site) so adding a new feature that utilizes that parameter is probably not a good idea.  Would probably make more sense to be a separate web pref that gets propogated to the node env when we initialize it in the renderer.

Reply:
Turned into `webPreferences.frozenIntrinsics`